tests/ui/deprecations.stderr

```diff
@@ -10,6 +10,12 @@ note: the lint level is defined here
 1  | #![deny(deprecated)]
    |         ^^^^^^^^^^
 
+error: use of deprecated function `pyo3::deprecations::deprecate_implicit_option`: Implicit default for trailing optional arguments is phased out. Add an explicit `#[pyo3(signature = (...))]` attribute a to silence this warning. In a future pyo3 version `Option<..>` arguments will be treated the same as any other argument.
+   --> tests/ui/deprecations.rs:119:39
+    |
+119 | fn pyfunction_option_2(_i: u32, _any: Option<i32>) {}
```

Comment:
It would be awesome if the macro could emit the right signature here instead of `(...)`.

I wonder, what happens if we made the macro deprecation expand to something like:

```rust
#[deprecated(note = "this function has implicit defaults for the trailing `Option<T>` argument `_any`. These implicit defaults are being phased out. Add `#[pyo3(signature = (_i, _any = None))]` to this function")]
const PYFUNCTION_OPTION_2_SIGNATURE: () = ();


// immediately use the deprecated symbol to trigger the deprecation warning
const _: () = PYFUNCTION_OPTION_2_SIGNATURE;
```

I think if we attached that to the span of the whole function, rather than just an individual argument, it'd be super clear to users how to upgrade.

Reply:
That would be really nice! I'll give it a try